From 48cad258a4c9a20efffc687138407d69cdca24bf Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:59 -0400
Subject: [ethernet] i40e: Use the new i40e_get_fd_cnt_all function in other places

Message-id: <1409658042-22689-119-git-send-email-sassmann@redhat.com>
Patchwork-id: 90312
O-Subject: [RHEL7.1 PATCH 118/281] i40e: Use the new i40e_get_fd_cnt_all function in other places
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

We have a function to calculate this, so update the code to use the function.

Change-ID: Ia345b6fe6ec7f0b2dcf1199471b0d0f959ad3908
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 89132783016ae770a5e1f1cc7660dc312e386a85)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 4cf2c9d..a333958 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4712,8 +4712,7 @@ void i40e_fdir_check_and_reenable(struct i40e_pf *pf)
      (pf->flags & I40E_FLAG_FD_SB_ENABLED))
   return;
  fcnt_prog = i40e_get_current_fd_count(pf);
- fcnt_avail = pf->hw.fdir_shared_filter_count +
-            pf->fdir_pf_filter_count;
+ fcnt_avail = i40e_get_fd_cnt_all(pf);
  if (fcnt_prog < (fcnt_avail - I40E_FDIR_BUFFER_HEAD_ROOM)) {
   if ((pf->flags & I40E_FLAG_FD_SB_ENABLED) &&
       (pf->auto_disable_flags & I40E_FLAG_FD_SB_ENABLED)) {
diff --git a/drivers/net/ethernet/intel/i40e/i40e_txrx.c b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
index 1ecd9aa..5a22187 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
@@ -451,9 +451,7 @@ static void i40e_fd_handle_status(struct i40e_ring *rx_ring,
 
   /* filter programming failed most likely due to table full */
   fcnt_prog = i40e_get_current_fd_count(pf);
-  fcnt_avail = pf->hw.fdir_shared_filter_count +
-             pf->fdir_pf_filter_count;
-
+  fcnt_avail = i40e_get_fd_cnt_all(pf);
   /* If ATR is running fcnt_prog can quickly change,
    * if we are very close to full, it makes sense to disable
    * FD ATR/SB and then re-enable it when there is room.
-- 
1.7.1